### PR TITLE
Finish up prerequisites

### LINKED ISSUE #257: Settings: Link to a meson example while mentioning it
See https://gitlab.gnome.org/Hofer-Julian/gtk-rs-book/-/issues/16

### PR DESCRIPTION
**Pull request by Hofer-Julian**
Fixes #257

### CODE REVIEW

File: .github/workflows/book.yml [hunk 1]
```diff
@@ -38,6 +38,8 @@ jobs:
 
       - name: Setup mdBook
         uses: peaceiris/actions-mdbook@v1
+        with:
+          mdbook-version: "latest"
```
Comment (bilelmoussaoui) [inline .github/workflows/book.yml:42]: latest is the default no?

Comment (Hofer-Julian) [reply .github/workflows/book.yml:42]: Mmmh, but I used the latest on Gitlab as well...

Then that might not be it.

File: book/src/prerequisites.md [hunk 2]
```diff
@@ -1,21 +1,73 @@
...
 
-# Buildsystem
+Now you can run your application by executing:
+```bash
+cargo run
+```
+
+## Cargo + Meson
+
+Cargo is *nearly* enough, but it is not well suited for handling resources such as icons or UI files.
```
Comment (bilelmoussaoui) [inline book/src/prerequisites.md:34]: I would add handling translation through gettext as well

File: .github/workflows/book.yml (hunk 1 above)
Comment (Hofer-Julian) [reply .github/workflows/book.yml:42]: I also tried it locally, and there this problem does not exist.

Maybe install mdbook with cargo for now?

Comment (Hofer-Julian) [reply .github/workflows/book.yml:42]: Or like the book does
https://github.com/rust-lang/book/blob/master/.github/workflows/main.yml

File: book/src/prerequisites.md [hunk 3]
```diff
@@ -1,21 +1,73 @@
...
+You only have to assure that Flatpak is installed on your system, check this [website](https://flatpak.org/setup/) to see if any steps are necessary on your distribution.
+Also download the [gtk-rust-template](https://gitlab.gnome.org/bilelmoussaoui/gtk-rust-template) and initialize it.
+
+Then either install:
+
+- [GNOME Builder](https://flathub.org/apps/details/org.gnome.Builder) or
+- [VSCodium](https://flathub.org/apps/details/com.vscodium.codium) together with the [rust-analyzer](https://open-vsx.org/extension/matklad/rust-analyzer) and [flatpak](https://open-vsx.org/extension/bilelmoussaoui/flatpak-vscode) extensions.
+
+That is it.
+Both IDEs will install all necessary dependencies for you, with GNOME Builder you only have to press the run button for that.
```
Comment (bilelmoussaoui) [inline book/src/prerequisites.md:71]: That's not actually true, only Builder is capable of doing so. Unless you meant the application dependencies

File: book/src/prerequisites.md (hunk 2 above)
Comment (Hofer-Julian) [reply book/src/prerequisites.md:34]: Kind of included in "GTK functionality".

I can replace "GTK functionality" with actual examples, but I wonder if that might be confusing for total GTK beginners

File: book/src/prerequisites.md (hunk 3 above)
Comment (Hofer-Julian) [reply book/src/prerequisites.md:71]: I thought runtime is installed by the extension or is this only planned?

Comment (bilelmoussaoui) [reply book/src/prerequisites.md:71]: It was never the case? planned, yes, no idea when i will get the time to look at it. Also, currently, only the GNOME 40 runtime contains gtk4, so you can probably just be explicit about how to install both the rust sdk & gnome one. and we can remove that once both IDEs can handle installing the extensions just fine.

File: book/src/prerequisites.md (hunk 2 above)
Comment (bilelmoussaoui) [reply book/src/prerequisites.md:34]: I don't see any mention of "GTK functionality"? I wouldn't go too specific in details but still give few reasons of what Meson would simplify for the developer.

Comment (Hofer-Julian) [reply book/src/prerequisites.md:34]: Sure, which ones would you deem the most important?

File: .github/workflows/book.yml (hunk 1 above)
Comment (bilelmoussaoui) [reply .github/workflows/book.yml:42]: This can be dropped per the matrix conversation :-)

File: book/src/prerequisites.md (hunk 2 above)
Comment (bilelmoussaoui) [reply book/src/prerequisites.md:34]: It simplifies a bunch of things: 
- Translations: Simplify a bunch of stuff through the gettext module like translating appdata/desktop files or installing translations
- Building/installing GResources (UI files and all)
- Installing files such as icons

File: book/src/prerequisites.md [hunk 4]
```diff
@@ -1,21 +1,73 @@
-# Installation
+# Prerequisites
+
+There are a few recommended ways to set up your working station in order to develop `gtk-rs` applications.
```
Comment (sdroege) [inline book/src/prerequisites.md:3]:
```suggestion
There are a few recommended ways to set up your workstation in order to develop `gtk-rs` applications.
```

Or "development environment" maybe?

File: book/src/prerequisites.md [hunk 5]
```diff
@@ -1,21 +1,73 @@
-# Installation
+# Prerequisites
```
Comment (sdroege) [inline book/src/prerequisites.md:1]: Should this include a paragraph about getting GTK4 and installing the Rust toolchain (e.g. via rustup)?

Comment (sdroege) [reply book/src/prerequisites.md:1]: I mean, separate from the `cargo` / `cargo + meson` sections

File: book/src/prerequisites.md (hunk 2 above)
Comment (sdroege) [reply book/src/prerequisites.md:34]: Also settings, the schemas

File: book/src/prerequisites.md [hunk 6]
```diff
@@ -1,21 +1,73 @@
...
+Now you can run your application by executing:
+```bash
+cargo run
+```
+
+## Cargo + Meson
+
+Cargo is *nearly* enough, but it is not well suited for handling resources such as icons or UI files.
+That is why we recommend to use [Meson](https://mesonbuild.com/) on top of it.
+It is cross-platform, its syntax is very readable and it offers many convenience functionalities for GTK.
```
Comment (sdroege) [inline book/src/prerequisites.md:36]:
```suggestion
It is cross-platform, its syntax is very readable and it offers many convenience features for GTK.
```

File: book/src/prerequisites.md (hunk 5 above)
Comment (Hofer-Julian) [reply book/src/prerequisites.md:1]: I thought about that. 
However, they are only two sentences and I am afraid that the overview gets worse if I put this separate.

**Comment by Hofer-Julian** [comment]
Should be fine now

File: book/src/prerequisites.md [hunk 7]
```diff
@@ -1,21 +1,76 @@
...
+```
+
+## Cargo + Meson
+
+Cargo is *nearly* enough, but it is not well suited for handling resources such as icons or UI files.
+That is why we recommend to use [Meson](https://mesonbuild.com/) on top of it.
+It is cross-platform and its syntax is very readable.
+Meson takes care of
+- translations
+- building and installing GResources
```
Comment (bilelmoussaoui) [inline book/src/prerequisites.md:39]: Could use a link to what a GResource is https://developer.gnome.org/gio/stable/GResource.html

File: book/src/prerequisites.md [hunk 8]
```diff
@@ -1,21 +1,76 @@
...
+- installing auxiliary files such as icons and schemas
```
Comment (bilelmoussaoui) [inline book/src/prerequisites.md:40]: Same for the schemas here https://developer.gnome.org/GSettings/ I guess

File: book/src/prerequisites.md (hunk 7 above)
Comment (Hofer-Julian) [reply book/src/prerequisites.md:39]: I think, I'll link to the corresponding chapter in the book instead.
Good point though :)

**Pull request merged**